prow/cmd/deck/main.go

```diff
@@ -345,17 +345,15 @@ func (c *filteringProwJobLister) ListProwJobs(selector string) ([]prowapi.ProwJo
 
 	var filtered []prowapi.ProwJob
 	for _, item := range prowJobList.Items {
-		if item.Spec.Refs == nil && len(item.Spec.ExtraRefs) == 0 {
-			// periodic jobs with no refs cannot be filtered
-			filtered = append(filtered, item)
-			continue
-		}
-
 		refs := item.Spec.Refs
 		if refs == nil {
-			refs = &item.Spec.ExtraRefs[0]
+			if len(item.Spec.ExtraRefs) > 0 {
```

Comment:
The fact that only the first `ExtraRef` is considered does seem to be a bit unexpected

Reply:
Yeah lets make the check more complete and consider `Spec.Refs` and all `Spec.ExtraRefs`.

Reply:
Will create a follow-up :+1: